### DISCUSSION TITLE
API matching MLS service interface

### DISCUSSION THREAD

**Discussion by snormore** [xmtp/xps-gateway#32] [Ideas and Features]
Our MLS delivery service interface is defined in protobufs [here](https://github.com/xmtp/proto/blob/snor/mls-api-query/proto/mls/api/v1/mls.proto) and implemented in our golang node [here](https://github.com/xmtp/xmtp-node-go/blob/snor/mls-msgs/pkg/mls/api/v1/service.go). This is basically what we need to build towards in terms of functionality, and so we should probably try to match the gateway interface with it.

The interface breaks down as follows:
- Contacts/registry
  - RegisterInstallation
  - UploadKeyPackage
  - FetchKeyPackages
  - RevokeInstallation
  - GetIdentityUpdates
- Group/conversation messaging
  - SendGroupMessages
  - QueryGroupMessages
  - SubscribeGroupMessages
- Welcome/invite messaging
  - SendWelcomeMessages
  - QueryWelcomeMessages
  - SubscribeWelcomeMessages

**Comment by jac18281828** [comment]
Yes, this makes sense.  I can migrate our existing tickets to use these names.
